docs/en/development-and-contribution/development-guide.md

```diff
@@ -0,0 +1,333 @@
+```
+
+#### Interceptor
+
+Interceptors are used to define custom business logic before and after method execution, 
+allowing you to access data from before and after method execution and interact with the Agent Core by using the Agent API.
+
+The interceptor definition is as follows, you need to create a new structure and implement it:
+
+```go
+type Interceptor interface {
+    // BeforeInvoke would be called before the target method invocation.
+    BeforeInvoke(invocation *Invocation) error
+    // AfterInvoke would be called after the target method invocation.
+    AfterInvoke(invocation *Invocation, result ...interface{}) error
+}
+```
+
+Within the interface, you can see the `Invocation` struct, which defines the context of an interception. The specific definition is as follows:
+
+```go
+type Invocation struct {
+	// CallerInstance is the instance of the caller, nil if the method is static method.
+	CallerInstance interface{}
+	// Args is the arguments data of the method, please cast to the specific type to get more inforamtion.
+	Args           []interface{}
+
+	// Continue is the flag to control the method invocation, if it is false, the target method would not be invoked.
+	Continue bool
+	// Return is the return data of the method, the result must be provided when continue the target method invoke.
+	// and the result size must be the same as the method return value size.
+	Return   []interface{}
+
+	// Context is the customized context of the method invocation, it should be propagated the tracing span.
+	Context interface{}
+}
+```
+
+##### Thread safe
+
+The `Interceptor` instance would **define new instance at the current package level**, 
+rather than creating a new instance each time a method is intercepted. 
+
+Therefore, please try not to define intercepted objects in the interceptor, and instead use `Invocation.Context` to pass data.
+
+##### Package Path
+
+If the method you want to intercept is not located in the root directory of the framework, 
+place your interceptor code in the relative location within the plugin. **The Agent would only copy files from the same package directory**.
+
+For example, if you want to intercept a method in `github.com/gin-gonic/gin/render`, create a **render** directory in the root of your plugin, and **put the interceptor inside it**. 
+This ensures that the interceptor is properly included during the copy operation and can be correctly applied to the target package.
+
+## Agent API
+
+The Agent API is used when a method is intercepted and interacts with the Agent Core.
+
+### Tracing API
+
+The Tracing API is used for building distributed tracing, and currently supports the following methods:
+
+```go
+// CreateEntrySpan creates a new entry span.
+// operationName is the name of the span.
+// extractor is the extractor to extract the context from the carrier.
+// opts is the options to create the span.
+func CreateEntrySpan(operationName string, extractor Extractor, opts ...SpanOption)
```

Comment:
I think all these comments are not available in the source codes, please copy to there too.